daft/dataframe/dataframe.py

```diff
@@ -549,62 +549,109 @@ def write_iceberg(self, table: "pyiceberg.table.Table", mode: str = "append") ->
         # This is due to the fact that the logical plan of the write_iceberg returns datafiles but we want to return the above data
         return with_operations
 
+    @DataframePublicAPI
     def write_delta(
         self,
-        path: str,
-        mode: str = "append",
+        table: Union[str, pathlib.Path, "deltalake.DeltaTable"],
```

Comment:
This should also be able to take in a `daft.io.catalog.DataCatalogTable` which lets users use their unity catalog